Add statistics support for Aurora integration

emiltorp opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Added `state_class` to be able to use the statistics cards

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

emiltorp committed:
Add statistics support for Aurora integration

mib1185 committed:
Replace the usage of unit constants by enumerations in Tests [m-n] (#85935)

* replace unit conts by enums m-n

* fix mqtt

mkmer committed:
Address invalid keys in translation for whirlpool (#85849)

DCSBL committed:
Implement state error handling in HomeWizard (#84991)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

frenck committed:
Make translations keys check hassfest more strict (#85221)

thecode committed:
Cleanup webOS TV YAML import leftovers (#85957)

mib1185 committed:
Replace the usage of unit constants by enumerations in Tests [v-z] (#85938)

replace unit conts by enums v-z

frenck committed:
Clean up legacy import from HomeWizard (#85960)

thecode committed:
Update webOS TV codeowners (#85959)

jbouwh committed:
Use value of enum (#85944)

DCSBL committed:
Upgrade HomeWizard to platinum quality (#82580)

frenck committed:
Core code styling improvements (#85963)

janiversen committed:
Bump pymodbus library to V3.1.0 (#85961)

fixes undefined

shbatm committed:
Bump PyISY to 3.1.6 (#85974)

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Ensure remote bluetooth adapters are loaded before integrations that need them (#85723)

mib1185 committed:
Make Synology DSM integration fully async (#85904)

Commit:
[ci skip] Translation update

shbatm committed:
Support availability for ISY994 devices (#85928)

shbatm committed:
Deprecate ISY994 custom cleanup entities service (#85931)

boswelja committed:
Add Starlink reboot button (#85729)

jesserockz committed:
Add friendly name support for ESPHome (#85976)

Co-authored-by: J. Nick Koston <nick@koston.org>

Nardol committed:
Add locking and unlocking to MQTT lock (#85779)

* Implement locking, unlocking and jammed on MQTT lock

Signed-off-by: Patrick ZAJDA <patrick@zajda.fr>

* Add tests

Signed-off-by: Patrick ZAJDA <patrick@zajda.fr>

* Refactor condition

Signed-off-by: Patrick ZAJDA <patrick@zajda.fr>

* Parametrize tests

Signed-off-by: Patrick ZAJDA <patrick@zajda.fr>

* Manage only locking and unlocking

Signed-off-by: Patrick ZAJDA <patrick@zajda.fr>

* Remove jammed from abbreviations

Co-authored-by: Jan Bouwhuis <jbouwh@users.noreply.github.com>

* set valid states in self._valid_states

Signed-off-by: Patrick ZAJDA <patrick@zajda.fr>

Signed-off-by: Patrick ZAJDA <patrick@zajda.fr>
Co-authored-by: Jan Bouwhuis <jbouwh@users.noreply.github.com>

mib1185 committed:
Replace the usage of unit constants by enumerations in Tests [s-u] (#85937)

bdraco committed:
Fix fetching history include_start_time_state when timezone is not UTC (#85983)

Jorei committed:
Add state_class to mill Estimated CO2 sensor (#85894)

frenck committed:
Update python-homewizard-energy to 1.5.0 (#85966)

* Update python-homewizard-energy to 1.5.0

* Remove strict typing for now

* Revert "Remove strict typing for now"

This reverts commit ebcd327fdf3e10abcaabd6cd86a0bfc101b32fb1.

* Adjust typing to resolve upstream changes

emontnemery committed:
Remove sky connect config entry if USB stick is not plugged in (#85765)

* Remove sky connect config entry if USB stick is not plugged in

* Tweak cleanup

* Give some stuff more cromulent names

* Do the needful

* Add tests

* Tweak

frenck committed:
Update pytest to 7.2.1 (#85990)

DCSBL committed:
Add sensors for new P1 datapoints in HomeWizard (#85198)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Franck Nijhof <git@frenck.dev>
fixes undefined

epenet committed:
Warn if numeric sensors have an invalid value (#85863)

Co-authored-by: mib1185 <mail@mib85.de>

emontnemery committed:
Make the kitchen_sink integration set up a config entry (#85680)

* Make the kitchen_sink integration set up a config entry

* Update homeassistant/components/kitchen_sink/config_flow.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Add singleton check in import step + add test

* Fix tests

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

frenck committed:
Improve typing of HomeWizard sensors (#85997)

* Improve typing of HomeWizard sensors

* Fix typo when tried to fix a typo :)

epenet committed:
Allow None in add_suggested_values_to_schema (#85763)

frenck committed:
Cleanup of HomeWizard button platform (#85999)

thecode committed:
Remove WebOS TV client wrapper (#86001)

frenck committed:
Clean up unused import steps from HomeWizard config flow (#86002)

lemanjo committed:
Bump pykoplenti to 1.0.0 (#83251)

* Update Kostal integration to use maintained lib

* Update Kostal integration to use pykoplenti

* Update kostal_plenticore tests for new lib

* Fix tests config_flow & diagnostics after changes

epenet committed:
Add missing mock in fritzbox (#86005)

epenet committed:
Adjust invalid test values in rest (#86009)

epenet committed:
Adjust invalid test values in snmp (#86006)

frenck committed:
Add sensor state class validation for device classes (#84402)

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

epenet committed:
Adjust invalid test values in tcp (#86007)

epenet committed:
Adjust invalid test values in template (#86008)

epenet committed:
Remove invalid uom from mqtt tests (#86004)

janiversen committed:
Bring modbus back to 100% test coverage (#85972)

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

epenet committed:
Adjust sensor validation warning (#86017)

piitaya committed:
Add tier summation delivered for Lixee Zlinky TIC (#82602)

* Add tier summation delivered for zlinky

* Improve name case

* Add other tiers and register tier

* Fix smartenergy sensor update

* Account for new reporting configuration in unit tests

* Use cluster ID attributes instead of hardcoding the values

* Use tier names instead of the numeric constants for formatter

* Revert active register tier delivered

* Fix tests

Co-authored-by: puddly <32534428+puddly@users.noreply.github.com>

TheJulianJES committed:
Implement "group members assume state" option for ZHA (#84938)

* Initial "group members assume state" implementation for ZHA

* Remove left-over debug flag (where polling was disabled)

* Implement _send_member_assume_state_event() method and also use after turn_off

* Only assume updated arguments from service call to group

* Make code more readable and change checks slightly

* Move "send member assume state" events to LightGroup on/off calls

* Include new config option in tests

* Check that member is available before updating to assumed state

* Lower "update group from child delay" for debouncer to basically 0 when using assumed member state

* Allow "child to group" updates regardless of config option

This is not needed, as group members will not update their state, as long as they're transitioning. (If a group transitions, it also sets its members to transitioning mode)

This fixes multiple issues. Previously, the state of a group was completely wrong when:
- turn on group with 10 second transition
- turn on members individually
- turn off members individually
- group state would not update correctly

* Move "default update group from child delay" constant

* Change to new constant name in test

* Also update fan test to new constant name

* Decrease "update group from child delay" to 10ms

In my testing, 0.0 also works without any issues and correctly de-bounces child updates when using the "assume state option".
This is just for avoiding multiple state changes when changing the group -> children issue individual updates.
With 2 children in a group and delay 0, both child updates only cause one group re-calculation and state change.

0.01 (10ms) should be plenty for very slow systems to de-bounce the update (and in the worst case, it'll cause just another state change but nothing breaks)

* Also implement "assuming state" for effect

Not sure if anybody even uses this, but this one is a bit special because the effect is always deactivated if it's not provided in the light.turn_on call.

* Move shortened delay for "assuming members" to a constant

* Add basic test to verify that group members assume on/off state

* Move _assume_group_state function declaration out of async_added_to_hass

* Fix rare edge-case when rapidly toggling lights and light groups at the same time

This prevents an issue where either the group transition would unset the transition flag or the single light would unset the group transition status midst-transition.

Note: When a new individual transition is started, we want to unset the group flag, as we actually cancel that transition.

* Check that effect list exists, add return type

* Re-trigger CI due to timeout

* Increase ASSUME_UPDATE_GROUP_FROM_CHILD_DELAY slightly

The debouncer is used when updating group member states either by assuming them (in which case we want to barely have any delay), or between the time we get the results back from polling (where we want a slightly longer time).
As it's not easily possible to distinguish if a group member was updated via assuming the state of the group or by the polling that follows, 50 ms seems to be a good middle point.

* Add debug print for when updating group state

* Fix issues with "off brightness" when switching between group/members

This fixes a bunch of issues with "off brightness" and passes it down to the members correctly.
For example, if a light group is turned off with a transition (so bulbs get their level set to 1), this will also set the "off brightness" of all individual bulbs to the last level that they were at.

(It really fixes a lot of issues when using the "member assume group state" option. It's not really possible to fix them without that.)

Furthermore, issues where polling was previously needed to get the correct state after "playing with transitions", should now get be resolved and get correct state when using the "members assume group state" option.

Note: The only case which still can't be fixed is the following:
If individual lights have off_with_transition set, but not the group, and the group is then turned on without a level, individual lights might fall back to brightness level 1 (<- at least now shows correctly in UI even before polling).
Since all lights might need different brightness levels to be turned on, we can't use one group call. But making individual calls when turning on a ZHA group would cause a lot of traffic and thereby be counter-productive.
In this case, light.turn_on should just be called with a level (or individual calls to the lights should be made).

Another thing that was changed is to reset off_with_transition/off_brightness for a LightGroup when a member is turned on (even if the LightGroup wasn't turned on using its turn_on method).
off_with_transition/off_brightness for individual bulbs is now also turned off when a light is detected to be on during polling.

Lastly, the waiting for polled attributes could previously cause "invalid state" to be set (so mid-transition levels).
This could happen when group and members are repeatedly toggled at similar times. These "invalid states" could cause wrong "off brightness" levels if transitions are also used.
To fix this, we check after waiting for the polled attributes in async_get_state to see if a transition has started in the meanwhile. If so, the values can be discarded. A new poll will happen later and if using the "members assume group state" config option, the values should already be correct before the polling.

* Enable "group members assume state" config option by default

The config tests are also updated to expect the config option be enabled by default.

For all tests, the config option is generally disabled though:
There are only two group related tests. The one that tests this new feature overrides the config option to be enabled anyway.
The other tests works in a similar way but also "sends" attribute reports, so we want to disable the feature for that test.
(It would also run with it enabled (if the correct CHILD_UPDATE value is patched), but then it would test the same stuff as the other test, hence we're disabling the config option for that test.)

epenet committed:
Avoid import homeassistant.const as a module (#85991)

frenck committed:
Small refactor to HomeWizard config flow (#86020)

* Small refactor to HomeWizard config flow

* Update homeassistant/components/homewizard/config_flow.py

Co-authored-by: Duco Sebel <74970928+DCSBL@users.noreply.github.com>

* Process review comments

Co-authored-by: Duco Sebel <74970928+DCSBL@users.noreply.github.com>

shbatm committed:
Fix entity cleanup and naming bugs in ISY994 (#86023)

thecode committed:
Bump aiowebostv to 0.3.2 (#86031)

fixes undefined

frenck committed:
Code styling tweaks to the recorder integration (#86030)

allenporter committed:
Add a timeout during OAuth token exchange and additional debug logging (#85911)

marcelveldt committed:
Remove SSDP discovery from Hue (#85506)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
Co-authored-by: Franck Nijhof <git@frenck.dev>

RunC0deRun committed:
Add Jellyfin media source support for tvshows (#85953)

mib1185 committed:
Remove the usage of HA core unit constant in withings own unit definitions (#85942)

shbatm committed:
Deprecate ISY994 custom reload service (#86019)

mib1185 committed:
Replace deprecated constants and utils by their successors in NWS tests (#85926)

jrieger committed:
Add Fire TV virtual integration (#85741)

shbatm committed:
Add Insteon backlight control support to ISY994, bump PyISY to 3.1.8 (#85981)

Co-authored-by: J. Nick Koston <nick@koston.org>

mvdwetering committed:
Set deviceclass to speaker for Sonos media_player (#80694)

shbatm committed:
Add enable/disable config switch for ISY994 devices (#85975)

Co-authored-by: J. Nick Koston <nick@koston.org>

mxr committed:
Run `flake8` on more files (#85333)

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>
Co-authored-by: Dave T <17680170+davet2001@users.noreply.github.com>

mib1185 committed:
Make central AvmWrapper class fully async in Fritz!Tools (#83768)

natekspencer committed:
Code cleanup in litterrobot (#86037)

karliemeads committed:
Fix logic of disabled condition for "OR" (#79718)

matteocorti committed:
Bump openerz-api to 0.2.0 (#86013)

Co-authored-by: Franck Nijhof <git@frenck.dev>
fixes undefined

epenet committed:
Set onewire quality scale to gold (#85748)

tyge68 committed:
Add Tuya sensor for Smart IR remote device (#85955)

rsmatias committed:
Add switch to Tuya product category wsdcg (#85762)

emontnemery committed:
Deduplicate some entity registry code (#85541)

Danielhiversen committed:
Update pyTibber to 0.26.8 (#86044)

kamiyo committed:
Improve type hints in nws (#83173)

frenck committed:
Update orjson to 3.8.5 (#86043)

bdraco committed:
Handle ignored shelly entries when discovering via zeroconf (#86039)

fixes https://github.com/home-assistant/core/issues/85879

iMicknl committed:
Add SSHd and GH CLI to devcontainer to support `gh net` (#81623)

frenck committed:
Update whois to 0.9.23 (#86042)

Andre0512 committed:
Add some sensors and controls to tuya dehumidifier (#85380)

fixes undefined

scop committed:
Remove signal strength state class from Huawei LTE transmit power sensor (#85973)

Co-authored-by: Franck Nijhof <git@frenck.dev>

timmo001 committed:
Add typings to OVO Energy integration (#75944)

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>
Co-authored-by: Franck Nijhof <git@frenck.dev>

Commit:
[ci skip] Translation update

Lash-L committed:
oralb async_poll doc comment typo (#86049)

bdr99 committed:
Add EufyLife Bluetooth integration (#85907)

Co-authored-by: J. Nick Koston <nick@koston.org>

frenck committed:
Code styling tweaks to the zeroconf integration (#86048)

boswelja committed:
Address Starlink code quality and add update sensor (#86066)

* Use dt.now

* Bring back update binary sensor

* Don't patch coordinator

* Add silver quality scale

emontnemery committed:
Allow converting units of additional sensor device classes (#86072)

bdraco committed:
Make event.time_tracker_utcnow patchable with freezegun (#86051)

epenet committed:
Remove deprecated Renault service (#86070)

frenck committed:
Removes OpenALPR Local integration (#85544)

oischinger committed:
Add vicare buffer top/main temperature sensors (#79466)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
fixes undefined

hpirila committed:
Add sous vide start functionality (#84447)

epenet committed:
Raise HomeAssistantError in Renault (#86071)

epenet committed:
Set renault quality scale to platinum (#85753)

* Set renault quality scale to platinum

* Ensure coordinators do not run at the same time

* Add comment

epenet committed:
Fix invalid constant in vicare (#86079)

Shutgun committed:
Add diagnostics to devolo Home Network (#86022)

ratsept committed:
Soma connect update (#85682)

* Add support for Connect U1

* update pysoma to latest version

* Changes requested by epenet

* Extend exeption handling to pass all tests

emontnemery committed:
Add Thread integration (#85002)

* Add Thread integration

* Add get/set operational dataset as TLVS

* Add create operational dataset

* Add set thread state

* Adjust after rebase

* Improve HTTP status handling

* Improve test coverage

* Change domains from thread to otbr

* Setup otbr from a config entry

* Add files

* Store URL in config entry data

* Make sure manifest is not sorted

* Remove useless async

* Call the JSON parser more

* Don't raise exceptions without messages

* Remove stuff which will be needed in the future

* Remove more future stuff

* Use API library

* Bump library to 1.0.1

lunmay committed:
Fix typo in energy: misplaced closing brace (#86027)

DCSBL committed:
Add Enum device class for HomeWizard active tariff (#86078)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

emontnemery committed:
Remove config entry specifics from FlowManager (#85565)

marcelveldt committed:
Fix Matter unique_id generation (#86046)

* bae entity unique id on Operational Instance Name standard

* Update homeassistant/components/matter/entity.py

Co-authored-by: Stefan Agner <stefan@agner.ch>

* also adjust unique id for devices

* final adjustment

* remove assert on server_info

* move device info to init

* fabric_id_hex

* use DeviceInfo instead of dict

* fix test

Co-authored-by: Stefan Agner <stefan@agner.ch>

emontnemery committed:
Use Home Assistant swing modes in tado climate (#84278)

jbouwh committed:
Allow translating select selector options (#85531)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

joncar committed:
Bump pylitejet to 0.4.6 (now with async!) (#85493)

emontnemery committed:
Allow manually setting up the Thread integration (#86087)

frenck committed:
Code styling tweaks to the Matter integration (#86096)

emontnemery committed:
Adjust Thread config flow (#86097)

* Adjust Thread config flow

* Improve tests

* Update homeassistant/components/otbr/config_flow.py

Co-authored-by: Franck Nijhof <git@frenck.dev>

Co-authored-by: Franck Nijhof <git@frenck.dev>

bachya committed:
Subclass a `DataUpdateCoordinator` for Ridwell (#85644)

emontnemery committed:
Adjust kitchen_sink test (#86099)

emontnemery committed:
Improve logging shutdown (#85812)

* Improve logging shutdown

* Update test

elupus committed:
Adjust translations for zha buttons (#86095)

bdraco committed:
Bump govee-ble to 0.21.1 (#86103)

fixes https://github.com/home-assistant/core/issues/85580

killer0071234 committed:
Remove deprecated zamg YAML config (#86113)

bdraco committed:
Ensure flux_led does not try to reload while setup is in progress (#86122)

There was a discovery race in flux_led that could trigger
it to try to reload while in progress if the device data
changed

frenck committed:
Code styling tweaks to the Alexa integration (#86121)

GitHub-Action committed:
[ci skip] Translation update

bdraco committed:
Fix live logbook stalling when there are no historical events with a high commit interval (#86110)

* Force live logbook to send an empty message to indicate no results

Since the sync task can take a while if the recorder is
busy, the logbook will appear to hang if we do not send
the first partial message even if its empty.

This work is in preparation for a higher database
commit interval where this issue is most obvious.

The historical only path did not have this issue because
it never had to wait for the db sync.

* update tests

mdonoughe committed:
Update to pylutron_caseta to 0.18.0 (#86133)

mkmer committed:
Bump AIOAladdinConnect to 0.1.53 (#86129)

GrahamJB1 committed:
Reset Modbus value on down (#86127)

* modbus: slave should reset value on sensor down as parent does that

* modbus: slave should reset value on sensor down as parent does that

thecode committed:
Remove WebOS TV script translation leftover (#86109)

Keep only english changes

emontnemery committed:
Only wait for import flows to initialize at setup (#86106)

* Only wait for import flows to initialize at setup

* Update hassio tests

* Update hassio tests

* Apply suggestions from code review

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

frenck committed:
Code styling tweaks to the ESPHome integration (#86146)

Drafteed committed:
Bump pybravia to 0.3.1 (#86153)

fixes undefined

epenet committed:
Fix incorrect type hint in Filter (#86141)

epenet committed:
Rename precision variable in Filter (#86090)

cdce8p committed:
Add missing raise for exceptions (#86155)

cdce8p committed:
Remove unnecessary try-else (4) (#86161)

cdce8p committed:
Remove unnecessary try-else (5) (#86164)

cdce8p committed:
Remove unnecessary try-else (1) (#86158)

cdce8p committed:
Remove unnecessary try-else (2) (#86159)

cdce8p committed:
Remove unnecessary try-else (3) (#86160)

emontnemery committed:
Change 'Sky Connect' to 'SkyConnect' (#86166)

cdce8p committed:
Update pylint to 2.15.10 (#86167)

mkmer committed:
Change Honeywell somecomfort API to AIOSomecomfort API (#86102)

* Move to AIOSomecomfort

* Remove unused constant

* Improve test coverage to 100

* Update homeassistant/components/honeywell/__init__.py

remove "todo" from code

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Missing cannot_connect translation

* add asyncio errors
update devices per entity
rework retry login

Co-authored-by: Erik Montnemery <erik@montnemery.com>

marcelveldt committed:
Adjust device registry for Matter devices (#86108)

* adjust device registry

* ignore test unique id

* update test

* ditch uniqueid + prefix serial

* adjust test

* add tests

* fix switch test

* prefix all identifiers

* Update homeassistant/components/matter/adapter.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* no underscore in id

* fix test

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Kane610 committed:
Use more _attrs_* in Axis entities (#85555)

* Use _attr_available

* Use _attr_is_on

* Use _attr_name

* Make some values private

* Update names of axis entity base classes

* Fix review comments

balloob committed:
Add OTBR WebSocket API (#86107)

* Add OTBR WebSocket API

* Not always active dataset

* Move logic to data class

* Remove retry until we need it

* Test all the things

thecode committed:
Add reauth flow to webOS TV integration (#86168)

* Add reauth flow to webOS TV integration

* Remove unnecessary else

balloob committed:
Reload ESPHome config entries when dashboard info received (#86174)

930913 committed:
Add sensor platform to LD2410BLE (#85276)

* Add sensor platform to LD2410BLE

- Add platform
- Add moving target distance entity
- Add static target distance entity
- Add moving target energy entity
- Add static target energy entity

* Add detection distance entity

* Align bluetooth-data-tools version

* Generate sensors from description

Also add state_class and unfactor description lambdas.

* Optimise LD2410BLE collections

Co-authored-by: J. Nick Koston <nick@koston.org>

balloob committed:
ESPHome dashboard tweaks (#86176)

bdr99 committed:
Rename Eufy integration to EufyHome (#86065)

thecode committed:
Fix Shelly sleeping Gen2 device updates (#86198)

teharris1 committed:
Cleanup Insteon code issues (#86173)

Clean up code issues

GitHub-Action committed:
[ci skip] Translation update

boswelja committed:
Add a switch to Starlink for stow/unstow (#85730)

Co-authored-by: J. Nick Koston <nick@koston.org>

synesthesiam committed:
Add conversation reload service (#86175)

* Add preload and reload service calls to conversation

* Add conversation preload/reload to websocket API

* Merge prepare into reload

* reload service and prepare websocket API

* Add preload and reload service calls to conversation

* Add conversation preload/reload to websocket API

* Merge prepare into reload

* reload service and prepare websocket API

* Add language lock for loading intents

* Add more tests for code coverage

epenet committed:
Add type hints to Filter integration tests (#86169)

* Add type hints to filter tests

* Adjust

* Ensure strings are passed to State constructor

* Simplify Recorder import

epenet committed:
Add type hints to Filter (#86165)

frenck committed:
Clean up HomeWizard diagnostic tests (#86211)

GrahamJB1 committed:
Support float in modbus register sensor (#86128)

register sensor should support float

Gerigot committed:
75142 Added QEMU_CPU ARG to the dockerfile (#86178)

at the moment when building for armhf its downloading the wrong packages since version 2022.7.6
Trying to start newer version with an armv6 raspberry leads to an infinite loop of exit code 256

epenet committed:
Add filter to strict-typing (#86215)

* Add filter to strict-typing

* Adjust comment

GrahamJB1 committed:
modbus slave unique ids (#86126)

modbus slave unique ids

akx committed:
Fix stray string literal in bluetooth test wrapper (#86228)

akx committed:
Fix docstring in esphome.bluetooth.client (#86226)

balloob committed:
Add conversation mobile app webhook (#86239)

* Add conversation mobile app webhook

* Re-instate removed unused import which was used as fixture

jpelgrom committed:
Update allowlisted OAuth redirect URIs for Wear OS in China (#86247)

balloob committed:
Update ESPHome devices from HA (#86249)

* Update ESPHome devices from HA

* esphome-dashboard-api==1.2.2

* Limit to 1 parallel ESPHome update

synesthesiam committed:
HassTurnOn/Off intents to also handle cover entities (#86206)

* Move entity/area resolution to async_match_states

* Special case for covers in HassTurnOn/Off

* Enable light color/brightness on areas

* Remove async_register from default agent

* Remove CONFIG_SCHEMA from conversation component

* Fix intent tests

* Fix light test

* Move entity/area resolution to async_match_states

* Special case for covers in HassTurnOn/Off

* Enable light color/brightness on areas

* Remove async_register from default agent

* Remove CONFIG_SCHEMA from conversation component

* Fix intent tests

* Fix light test

* Fix humidifier intent handlers

* Remove DATA_CONFIG for conversation

* Copy ServiceIntentHandler code to light

* Add proper errors to humidifier intent handlers

GitHub-Action committed:
[ci skip] Translation update

cdce8p committed:
Improve `ld2410_ble` generic typing (#86258)

akx committed:
Fix docstring in helpers.template_entity (#86227)

klaasnicolaas committed:
Bump odp-amsterdam to v5.0.1 (#86252)

Bump package version to v5.0.1

DCSBL committed:
Bumb python-homewizard-energy to 1.6.0 (#86255)

starkillerOG committed:
Bump reolink-aio to 0.3.0 (#86259)

* Bump reolink-aio to 0.3.0

* fix typo

* ReolinkException

thecode committed:
Shelly - use common coordinator base class (#86262)

* Shelly - use common coordinator base class

* rename entry to device_entry

Co-authored-by: J. Nick Koston <nick@koston.org>

Co-authored-by: J. Nick Koston <nick@koston.org>

cdce8p committed:
Add option to run only pylint or mypy tests [ci] (#86260)

jbouwh committed:
Add jammed state support for MQTT lock (#86010)

* Add jammed state support for MQTT lock

* Correct payload jammed key

* Add tests - rename solved to ok

* Rename jammed state and template topics to motor

* Use state topic for handling motor state

* Follow up comments

* Change default behaviour `state_unjammed`

* Skip `state_unjammed`

mib1185 committed:
Support password less PI-Hole installations (#86184)

fixes undefined

DCSBL committed:
Add sensors for smart and gas meter identifiers (serial numbers) in HomeWizard (#86282)

fwestenberg committed:
Add Stookwijzer (#84435)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Franck Nijhof <git@frenck.dev>

frenck committed:
Stricter pylint message control (#86154)

frenck committed:
Code styling tweaks to the tests - Part 1 (#86192)

emontnemery committed:
Import recorder locally in test fixtures (#86286)

emontnemery committed:
Rename `otbr` integration (#86284)

Rename otbr integration

frenck committed:
Add battery device class to Glances battery sensor (#86278)

frenck committed:
Automatically set up HomeWizard during onboarding (#86024)

* Automatically set up HomeWizard during onboarding

* Add disabled API during onboarding test

* Mark onboarding complete half way in the test

frenck committed:
Update python-homewizard-energy to 1.6.1 (#86274)

* Update python-homewizard-energy to 1.6.1

* Adjust tests

frenck committed:
Refactor HomeWizard switch platform to use entity descriptions (#86011)

frenck committed:
Remove deprecated Uptime YAML configuration (#86292)

thecode committed:
Fix Shelly sleeping Gen2 - do not refresh from zeroconf discovery (#86296)

mkmer committed:
Address Honeywell late review (#86202)

azogue commented on homeassistant/components/aurora/sensor.py:
```diff
@@ -33,3 +33,8 @@ class AuroraSensor(AuroraEntity, SensorEntity):
+    @property
+    def state_class(self):
+        """Return the state class of this entity, from STATE_CLASSES, if any."""
+        return SensorStateClass.MEASUREMENT
```
Hi @emiltorp, the `state_class` property is not used anymore when it's constant. Better to add it as `_attr_state_class = SensorStateClass.MEASUREMENT`, near the current `_attr_native_unit_of_measurement` 👍

emiltorp replied:
Thanks. Good suggestion, updated the code and tested that it still works.

starkillerOG committed:
Reolink add binary sensors (#85654)

Co-authored-by: J. Nick Koston <nick@koston.org>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
Co-authored-by: Franck Nijhof <frenck@frenck.nl>

jbouwh committed:
Update stale docstring on demo lock platform (#86306)

Update stale docstring

bieniu committed:
Fix `state_class` for Accuweather `precipitation` sensor (#86088)

natekspencer committed:
Correct pet_weight sensor state class in litterrobot (#86320)

Danielhiversen committed:
Update pyTibber to 0.26.11 (#86316)

shbatm committed:
Bump PyISY to 3.1.9 for performance improvements (#86297)

emontnemery committed:
Improve device automation validation (#86143)

GitHub-Action committed:
[ci skip] Translation update

synesthesiam committed:
Conversation config (#86326)

* Restore conversation config

* Fall back to en for en_US, etc.

* Simplify config passing around

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

epenet committed:
Ensure all unit converters are tested (#86271)

* Ensure all unit converters are tested

* Adjust

* Simplify

* Simplify

* docstring

emiltorp committed:
Use constant _attr_state_class instead

emiltorp committed:
Use constant _attr_state_class instead

epenet committed:
Bump bcrypt to 4.0.1 (#86338)

Lash-L committed:
Fix incorrect mock in whirlpool (#86331)

* Added async according to error logs

* changed Asyncmock to magicmock get_aircon_mock

* changed attr_callback to MagicMock for sensor_mock

mkmer committed:
Fix preset modes in Honeywell  (#86293)

* Fix for issue #83841

* in instead of =

* Address None for entity maps

* Rework retry logic

* Committed to the wrong branch....
This reverts commit 40e19407a3 (Rework retry logic, 2023-01-21).

* Remove none, change log wording

mkmer committed:
Fix Honeywell unavailable state on connection lost (#86312)

* Fix for Issue 62957

* Cleanup exception test

* rework connection error retry logic

* Refactor HoneywellData class

* move _atr_available to correct location

* await create_task

AngellusMortis committed:
Fix edge cases for adding/enabling sensors for UniFi Protect (#86329)

Co-authored-by: J. Nick Koston <nick@koston.org>
fixes undefined

bdraco committed:
Add zeroconf discovery to Synology DSM (#86062)

bdraco committed:
Restore flume scan interval to match app (#86354)

To stay under the API limit the device connection is now
only checked hourly and notifications are only checked every
5 minutes

fixes #82336

bdraco committed:
Bump flux_led to 0.28.35 (#86352)

changelog: https://github.com/Danielhiversen/flux_led/compare/0.28.34...0.28.35

bdraco committed:
Switch an `asyncio.wait_for` in the template helper to `async_timeout` (#86349)

Switch an asyncio.wait_for in the template helper to async_timeout

Eliminates the extra task when calling async_render_will_timeout

elupus committed:
Don't interpret negative verbosity as debug (#86318)

* Don't interpret negative verbosity as debug

CI run with -qq which is a negative verbosity

* Make sure all caplog tests get debug level

* Update tests/conftest.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

Co-authored-by: Erik Montnemery <erik@montnemery.com>

bdraco committed:
Bump recommended esphome version for bluetooth proxies to 2022.12.4 (#86308)

This will fix an MTU issue reported with airthings and
other devices.

needs https://github.com/esphome/esphome/pull/4323

Shutgun committed:
React on IP changes in devolo Home Network (#86195)

thecode committed:
Cleanup Shelly imports (#86359)

* Cleanup Shelly imports

* Cleanup tests

GitHub-Action committed:
[ci skip] Translation update

thecode committed:
Revert "Add SSHd and GH CLI to devcontainer to support `gh net`" (#86360)

Revert "Add SSHd and GH CLI to devcontainer to support `gh net` (#81623)"

This reverts commit 15db63bb3b6fd9c0d35ec6d15783f1c5a96ada65.

bdraco committed:
Add support for streaming (push) history (#85892)

* Add support for streaming (push) history

Currently we poll for history data, like logbook we can stream
this data to avoid database overhead

* Update homeassistant/components/history/__init__.py

* merge filter

* expose new api

* expose new api

* expose new api

* expose new api

* coverage

* tests

* fixes

* tweak

* tweak

* tweak

* DRY

* leaky

* test for specific entities

* test for specific entities

* test for specific entities

* test for specific entities

* test for specific entities

* cover

* cover

* more cover

* tweak

* make sure it works before history starts

* fix test

* cover

* tweak

* make sure we unsub on overflow

* Update homeassistant/components/history/__init__.py

* Update homeassistant/components/history/__init__.py

* fix race in test

* fix db executor access

* relo above task creation

bdraco committed:
Avoid creating logbook stream task if unsubscribed while waiting for executor (#86363)

teharris1 committed:
Allow changing the Insteon USB device (#86290)

* Reconfig PLM device

* Test config change

bdraco committed:
Increase default recorder commit interval to 5 seconds (#86115)

bdraco committed:
Split history websocket API into its own file (#86364)

gjohansson-ST committed:
Bump pytrafikverket to 0.2.3 (#86341)

pytrafikverket 0.2.3

Co-authored-by: Shay Levy <levyshay1@gmail.com>

epenet committed:
Add missing ratio test cases in unit conversion (#86340)

* Add missing ratio test cases in unit conversion

* Improve ratio test to check inverse

DCSBL committed:
Upgrade python-homewizard-energy to 1.7.0 (#86383)

mib1185 committed:
Bump py-synologydsm-api to 2.0.2 (#86374)

mkmer committed:
Bump AIOSomecomfort to 0.0.3 (#86371)

fixes undefined

epenet committed:
Add per-file-ignore to pylint (#86289)

uvjustin committed:
Pass frag_duration as integer (#86375)

fixes undefined

Kane610 committed:
Use fixtures to setup Axis integration in tests (#86034)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bdraco committed:
Remove history use_include_order configuration option (#86365)

liudger committed:
Bump version python-bsblan to 0.5.9 (#86373)

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Fix repr for States and Events without a timestamp (#86391)

Kane610 committed:
Bump aiounifi to v44 (#86381)

fixes undefined

bdraco committed:
Bump yalexs-ble to 1.12.7 (#86396)

fixes https://github.com/home-assistant/core/issues/86182

frenck committed:
Drop Python 3.9 support (#85456)

MartinHjelmare committed:
Add mysensors remote platform (#86376)

starkillerOG committed:
Add Reolink dhcp discovery (#85880)

Co-authored-by: Franck Nijhof <git@frenck.dev>

thecode committed:
Shelly - handle None in RPC power sensors (#86399)

Handle None in RPC power sensors

thecode committed:
Cleanup Climacell translations (#86325)

jbouwh committed:
Improve MQTT transport select label (#86216)

daniele-athome committed:
Upgrade caldav to 1.0.1 (#85536)

Upgrade caldav (fix #40127)

date_search was deprecated in favour of search which also implements client-side recurring events expansion

GitHub-Action committed:
[ci skip] Translation update

Lash-L committed:
Change @conway20 to @Lash-L in OralB codeowners after github rename (#86416)

daniel-simpson committed:
Bump tesla_powerwall lib version to 0.3.19 (#86421)

zebardy committed:
Add support for additional fields to nut (#83265)

Co-authored-by: J. Nick Koston <nick@koston.org>

cdce8p committed:
Import `ParamSpec` from typing [Py310] (#86413)

* Import ParamSpec from typing [Py310]

* Update additional imports

cdce8p committed:
Update python version used for pylint [Py310] (#86414)

* Update python version used for pylint linting [Py310]

* Import Callable from collections.abc [Py310]

* Use builtin anext [Py310]

cdce8p committed:
Update Union typing (1) [Py310] (#86424)

cdce8p committed:
Update Union typing (2) [Py310] (#86425)

akx committed:
Bump pre-commit-hooks to 4.4.0 (#82984)

cdce8p committed:
Update Union typing (4) [Py310] (#86427)

cdce8p committed:
Update Union typing (5) [Py310] (#86428)

cdce8p committed:
Update Optional typing (2) [Py310] (#86419)

emiltorp committed:
Merge branch 'feature/addStatisticsToAurora' of github.com:emiltorp/core into feature/addStatisticsToAurora
